src/components/swap/SwapHeader.tsx

```diff
@@ -13,14 +22,136 @@ const StyledSwapHeader = styled.div`
+    error,
+    loading: fiatOnrampAvailabilityLoading,
+  } = useFiatOnrampAvailability(checkFiatRegionAvailability)
+
+  const disableBuyCryptoButton = Boolean(
+    error || (!fiatOnrampAvailable && fiatOnrampAvailabilityChecked) || fiatOnrampAvailabilityLoading
+  )
+
+  const [walletDrawerOpen, toggleWalletDrawer] = useWalletDrawer()
+
+  const handleBuyCrypto = useCallback(() => {
+    if (!fiatOnrampAvailabilityChecked) {
+      setCheckFiatRegionAvailability(true)
+      setContinueBuyFiatFlowPostRegionCheck(true)
+    } else if (fiatOnrampAvailable && !account && !walletDrawerOpen) {
+      toggleWalletDrawer()
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(true)
+    } else if (fiatOnrampAvailable && account) {
+      openFiatOnRampModal()
+      setBuyFiatClicked(true)
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(false)
+    } else if (!fiatOnrampAvailable) {
+      setBuyFiatClicked(true)
+      setContinueBuyFiatFlowPostRegionCheck(false)
+      setContinueBuyFiatFlowPostWalletModal(false)
+    }
+  }, [
+    walletDrawerOpen,
+    fiatOnrampAvailabilityChecked,
+    fiatOnrampAvailable,
+    account,
+    toggleWalletDrawer,
+    openFiatOnRampModal,
+    setBuyFiatClicked,
+  ])
+
+  useEffect(() => {
+    if (continueBuyFiatFlowPostRegionCheck || (account && continueBuyFiatFlowPostWalletModal)) {
+      handleBuyCrypto()
+    }
+  }, [
+    openFiatOnRampModal,
+    account,
+    checkFiatRegionAvailability,
+    fiatOnrampAvailabilityChecked,
+    fiatOnrampAvailable,
+    setBuyFiatClicked,
+    toggleWalletDrawer,
+    handleBuyCrypto,
+    setContinueBuyFiatFlowPostRegionCheck,
+    continueBuyFiatFlowPostWalletModal,
+    continueBuyFiatFlowPostRegionCheck,
+  ])
+
   return (
     <StyledSwapHeader>
       <RowBetween>
         <RowFixed>
-          <ThemedText.DeprecatedBlack fontWeight={500} fontSize={16} style={{ marginRight: '8px' }}>
+          <TextHeader className={subhead} color={theme.textPrimary}>
             <Trans>Swap</Trans>
-          </ThemedText.DeprecatedBlack>
+          </TextHeader>
+          {fiatOnRampButtonEnabled && (
+            <MouseoverTooltipContent
+              wrap={true}
+              content={
+                <div>
+                  <Trans>Crypto purchases are not available in your region. </Trans>
+                  <ExternalLink href={TOKEN_SAFETY_ARTICLE} style={{ paddingLeft: '4px' }}>
+                    <Trans>Learn more</Trans>
+                  </ExternalLink>
+                </div>
+              }
+              placement="bottom"
+              disableHover={!fiatOnrampAvailabilityChecked || fiatOnrampAvailable}
+            >
+              <TextHeader
+                className={subhead}
+                color={theme.textSecondary}
+                marginLeft="8px"
+                disabled={disableBuyCryptoButton}
+                isClickable={true}
```

Comment:
if `<TextHeader>` is always `isClickable` why is this prop here 🤔

Reply:
So TextHeader shouldn't always be clickable because Swap is also styled by TextHeader and it does nothing when clicked (thats the case where isClickable is false)